pkg/blockbuilder/blockbuilder_test.go

```diff
@@ -287,6 +280,121 @@ func TestBlockBuilder_WithMultipleTenants(t *testing.T) {
 	}
 }
 
+func TestBlockBuilder_WithOutOfOrderRecordsAndSamples(t *testing.T) {
```

Comment:
[nit] This test assumes ConsumeInterval is 1h, which is the default, but it may change in the future. To keep this test stable, I would suggest to override ConsumeInterval to 1h in the code, and add a comment to explain that's an assumption of the test, regardless what the default value will be in the future.

Should we do the same for ConsumeIntervalBuffer?